src/main/java/world/bentobox/bentobox/managers/IslandsManager.java

```diff
@@ -679,25 +680,23 @@ public void removePlayer(World world, UUID uuid) {
      * @param island to remove players from
      */
     public void removePlayersFromIsland(Island island) {
-        // Teleport players away
-        for (Player player : Bukkit.getOnlinePlayers()) {
-            if (island.inIslandSpace(player.getLocation().getBlockX(), player.getLocation().getBlockZ())) {
```

Comment:
This is actually a bug. It does not consider the world, so it could teleport players in other worlds! Oops.

Reply:
Good catch!